src/lib.rs

```diff
@@ -24,7 +26,8 @@ static INITIALIZED: Lazy<AtomicBool> = Lazy::new(|| AtomicBool::new(false));
 static SERVER: Lazy<RwLock<Option<Server>>> = Lazy::new(|| RwLock::new(None));
 
 struct Server {
-    ipc: IPC<Event>,
+    ipc_mission: IPC<Event>,
+    ipc_hook: IPC<()>,
```

Comment:
We now basically have two channels. One for the mission and one for the hook. This is how we don't need two actual TCP servers.

Reply:
That is exceedintly cool. Is the limit of calling into the mission env then shared between the two?

Reply:
Each env comes with its own limits for now. The main challenge for a shared limit would probably to figure out a way of distribution the calls in a fair manner between both (and to not create dead-locks if client calls to the mission env depend on a call to the hook env)